code/modules/hydroponics/gene_modder.dm

```diff
@@ -130,11 +152,37 @@
 			if("extract")
 				dat += "<span class='highlight'>[target.get_name()]</span> gene from \the <span class='highlight'>[seed]</span>?<br>"
 				dat += "<span class='bad'>The sample will be destroyed in process!</span>"
-				if(istype(target, /datum/plant_gene/core/potency))
+				if(istype(target, /datum/plant_gene/core))
 					var/datum/plant_gene/core/gene = target
-					if(gene.value > max_extract_pot)
-						dat += "<br><br>This device's extraction capabilities are currently limited to [max_extract_pot] potency. "
-						dat += "Target gene will be degraded to [max_extract_pot] potency on extraction."
+					if(istype(target, /datum/plant_gene/core/potency))
+						if(gene.value > max_potency)
+							dat += "<br><br>This device's extraction capabilities are currently limited to <span class='highlight'>[max_potency]</span> potency. "
+							dat += "Target gene will be degraded to <span class='highlight'>[max_potency]</span> potency on extraction."
+					if(istype(target, /datum/plant_gene/core/lifespan))
```

Comment:
all of these after the first one should be ```else if()```s, to avoid checking them all unnecessarily.